### PR TITLE
groth16: update api for circom format

### PR DESCRIPTION
**Pull request by joyqvq**
- match proof, aux_inputs, public_inputs as the same format as typescript struct definition to ensure bcs ser/de consistency. 
- change verify api to take in circom format directly and convert to arkwork struct. instaed of converting to arkworks bytes first and then arkworks struct.

### CODE REVIEW

File: fastcrypto-zkp/src/bn254/zk_login.rs [hunk 1]
```diff
@@ -341,7 +317,7 @@ impl AuxInputs {
         let mut poseidon = PoseidonWrapper::new(11);
         let jwt_sha2_hash_0 = Bn254Fr::from_str(&self.jwt_sha2_hash[0]).unwrap();
```
Comment (benr-ml) [inline fastcrypto-zkp/src/bn254/zk_login.rs:318]: are all those unwraps are safe? maybe use expect("reason") / document

File: fastcrypto-zkp/src/bn254/zk_login.rs [hunk 2]
```diff
@@ -492,58 +472,71 @@ impl ParsedMaskedContent {
...
     }
 
-    /// Get the all_inputs_hash bigint string.
-    pub fn get_all_inputs_hash(&self) -> &str {
-        &self.all_inputs_hash
+    /// Convert the public inputs into arkworks format.
+    pub fn as_arkworks(&self) -> Vec<Bn254Fr> {
+        self.inputs
+            .iter()
+            .map(|x| Bn254Fr::from_str(x).unwrap())
```
Comment (benr-ml) [inline fastcrypto-zkp/src/bn254/zk_login.rs:533]: is this always safe?

File: fastcrypto-zkp/src/bn254/zk_login.rs [hunk 3]
```diff
@@ -492,58 +472,71 @@ impl ParsedMaskedContent {
...
+            .collect()
     }
 
-    /// Get the serialized hash bytes.
-    pub fn get_serialized_hash(&self) -> &[u8] {
-        &self.serialized_hash
+    /// Get the all_inputs_hash as big int string.
+    pub fn get_all_inputs_hash(&self) -> &str {
+        &self.inputs[0]
```
[the remark this answers is not in the document]
Comment (joyqvq) [reply fastcrypto-zkp/src/bn254/zk_login.rs:539]: its defined from the backend prover response 
```
"[\"2487117669597822357956926047501254969190518860900347921480370492048882803688\"]"
```

File: fastcrypto-zkp/src/bn254/zk_login.rs (hunk 1 above)
Comment (joyqvq) [reply fastcrypto-zkp/src/bn254/zk_login.rs:318]: leaving as todo for now

**Review by mskd12 — APPROVED** [review]

**Pull request merged**